### PR TITLE
Add prettier

### PR DESCRIPTION
**Pull request by asmockler**
Adds prettier, per #1427. Supersedes #1896.

What this PR includes:
* Adds [`prettier`](https://prettier.io)
* Adds `.prettierrc` with project settings
* Set `formatOnSave` to `true` for VSCode users

What this PR _doesn't_ include:

* Formatting all files
* Add `eslint-plugin-prettier` and `eslint-config-prettier`
  * Since we are not formatting all files right away, these can't be added or else CI would fail. Would be good to, at some point, format everything and add these

@lukastaegert

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,4093 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (kellyselden) [inline yarn.lock]: This project uses npm, not yarn, for lockfiles

File: package.json [hunk 2]
```diff
@@ -101,5 +105,12 @@
     "typings/package.json.d.ts",
     "bin/rollup",
     "README.md"
-  ]
+  ],
+  "dependencies": {},
```
Comment (kellyselden) [inline package.json]: this can be removed

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 // Place your settings in this file to overwrite default and user settings.
 {
-  "typescript.format.insertSpaceBeforeFunctionParenthesis": true,
-  "typescript.format.insertSpaceAfterConstructor": true,
-  "typescript.format.enable": true
-}
\ No newline at end of file
+	"typescript.format.insertSpaceBeforeFunctionParenthesis": true,
```
Comment (kellyselden) [inline .vscode/settings.json]: can you preserve the formatting of this file please.

File: .prettierrc [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+	"singleQuote": true,
+	"useTabs": true
+}
```
Comment (lukastaegert) [inline .prettierrc]: 👍

File: yarn.lock (hunk 1 above)
Comment (asmockler) [reply yarn.lock]: 🤦‍♂️ d'oh. I was wondering why the diff on `yarn.lock` was so big...😅

**Comment by asmockler** [comment]
* :fire:d `yarn.lock` 🤦‍♂️ 
* Ran `npm install` to update `package-lock.json`
* Preserve formatting in `settings.json`
* Removed `formatOnSave: true` because it was running on all files, not just `*.ts`

**Comment by lukastaegert** [comment]
👍

File: package.json [hunk 5]
```diff
@@ -101,5 +105,11 @@
     "typings/package.json.d.ts",
     "bin/rollup",
     "README.md"
-  ]
+  ],
+  "lint-staged": {
```
Comment (azz) [inline package.json]: Hey, Prettier maintainer here. Awesome stuff. Just wanted to drop a line and ask if you've heard of [`pretty-quick`](https://github.com/azz/pretty-quick)? It makes this config a little bit cleaner.

Comment (lukastaegert) [reply package.json]: Hi, thanks for the tip, looks cool!

**Comment by guybedford** [comment]
This seems to be introducing quite a bit of noise into all commits now. Would it be possible to provide a single PR that does a full prettier conversion of the codebase? Would be a great help!

**Comment by guybedford** [comment]
Created https://github.com/rollup/rollup/pull/2031.

**Pull request merged**